tasks/build/os_shell_scripts.js

```diff
@@ -0,0 +1,41 @@
+import {unlink} from 'fs';
+import {join, extname} from 'path';
+import {promisify} from 'bluebird';
+import {ncp} from 'ncp';
+const pncp = promisify(ncp);
+const punlink = promisify(unlink);
+
+module.exports = function (grunt) {
+  grunt.registerTask('_build:osShellScripts', function () {
+    const done = this.async();
+    const source = 'build/kibana/bin';
+    const platforms = grunt.config.get('platforms');
+
+    Promise.resolve()
+      .then(() => allPlatforms(platform => punlink(join(platform.buildDir, 'bin')))
+      .then(() => allPlatforms(platform => pncp(source, join(platform.buildDir, 'bin'))))
+      .then(() => allPlatforms(removeExtraneousShellScripts)))
+      .then(() => done)
+      .catch(err => done(err));
+
+    function allPlatforms(fn) {
+      return Promise.all(platforms.map(fn));
+    }
+  });
+
+  function removeExtraneousShellScripts(platform) {
+    return Promise.all(grunt.file
+      .expand(join(platform.buildDir, 'bin', '*'))
+      .map(file => isExtraneous(file, platform) ?
```

Comment:
How about filtering on `isExtraneous` before mapping rather than manually resolving to undefined in those cases?

```js
.filter(file => isExtraneous(file, platform))
.map(punlink)
```